apiserver/adapters/in_memory_adapter.go

```diff
@@ -0,0 +1,59 @@
+package adapters
+
+import (
+	"errors"
+	"github.com/astanishevskyi/http-server/apiserver/models"
+	"math/rand"
+)
+
+type UserStorage struct {
+	UsersSlice []models.User `json:" users"`
+}
+
+func NewInMemoryUserStorage() *UserStorage {
+	storage := UserStorage{UsersSlice: make([]models.User, 0)}
+	return &storage
+}
+
+func (s *UserStorage) GetAll() []models.User {
+	return s.UsersSlice
+}
+
+func (s *UserStorage) Retrieve(id uint32) (models.User, error) {
+	for _, val := range s.UsersSlice {
+		if val.Id == id {
+			return val, nil
+		}
+	}
```

Comment:
What is the time complexity of this algorithm and what can be done to improve it?

Reply:
This algorithm has O(n) complexity. I have no idea how to simplify it, please help